main.cpp

```diff
@@ -56,46 +56,64 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const std::string c) : value(v), name(c) {}   //1
+    int value;
+    std::string name; 
 };
 
-struct <#structName1#>                                //4
+struct struct1                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if(a != nullptr && b != nullptr) 
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float name1 { 30 }, name2 { 20 };
+    float memberFunction(float* updatedValue)      //12
     {
-        
+        if(updatedValue != nullptr) 
+        {
+            std::cout << "U's name1 value: " << name1 << std::endl;
+            name1 = *updatedValue;
+            std::cout << "U's name1 updated value: " << name1 << std::endl;
+            while( std::abs(name2 - name1) > 0.001f )
+            {
+                name1 -= .5f;
+                name2 -= 1.0f;
+            }
+            std::cout << "U's name2 updated value: " << name2 << std::endl;
+            return name2 * name1;   
+        }
+        return 0;
     }
 };
 
-struct <#structname2#>
+struct struct2
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float staticFunctionA(U* that, float* updatedValue )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if(updatedValue != nullptr) 
```

Comment:
`that` is a pointer.  You need to check if `that` is null before you use it.
- [ ] revise